backend/database_handler/transactions_processor.py

```diff
@@ -366,3 +366,51 @@ def set_transaction_appeal_failed(self, transaction_hash: str, appeal_failed: in
             self.session.query(Transactions).filter_by(hash=transaction_hash).one()
         )
         transaction.appeal_failed = appeal_failed
+
+    def get_highest_timestamp(self) -> int:
+        transaction = (
+            self.session.query(Transactions)
+            .order_by(desc(Transactions.timestamp_accepted))
+            .first()
+        )
+        if transaction is None:
+            return 0
+        return transaction.nonce
+
+    def get_transactions_for_block(
+        self, block_number: int, include_full_tx: bool
+    ) -> dict:
+        transactions = (
+            self.session.query(Transactions)
+            .filter(Transactions.nonce == block_number)
```

Comment:
we should explain somehow that block_number === nonce

Reply:
this will forwarded to the Hardhat network in https://github.com/yeagerai/genlayer-studio/pull/755